LocalizationUtility/LocalizationUtility.cs

```diff
@@ -66,15 +89,123 @@ public void RegisterLanguage(ModBehaviour mod, string name, string translationPa
                 WriteLine($"Registering new language {name}");
 
                 TextTranslation.Language newLanguage = (hasAnyCustomLanguages ? _customLanguages.Values.Max(cl => cl.Language) : vanillaLanguages.Max()) + 1;
-                
-                _customLanguages[name] = new CustomLanguage(name, newLanguage, translationPath, mod, languageToReplace);
+
+                CustomLanguage customLanguage = new CustomLanguage(name,true, newLanguage, translationPath, mod, languageToReplace);
+                _customLanguages[name] = customLanguage;
+
+                TextTranslationPatches.AddNewTranslation(customLanguage, customLanguage.TranslationPath);
             }
             catch(Exception ex)
             {
                 WriteError($"Failed to register language. {ex}");
             }
         }
 
+        public void AddTranslation(ModBehaviour mod, string name, string translationPath)
+        {
+            try
+            {
+                WriteLine($"Adding translations to language {name}");
+
+                if (TryGetLanguage(name, out var customLanguage))
+                {
+                    TextTranslationPatches.AddNewTranslation(customLanguage, mod.ModHelper.Manifest.ModFolderPath + translationPath);
+                    return;
+                }
+                WriteError($"The custom language {name} isn't registered yet");
+
+            }
+            catch (Exception ex)
+            {
+                WriteError($"Failed to add translation to language {name}. {ex}");
+            }
+        }
+        #region NonXMLTranslationAdders
+        public void AddTranslation(string name, KeyValuePair<string, string>[] regularEntries, KeyValuePair<string, string>[] shipLogEntries, KeyValuePair<int, string>[] uiEntries)
+        {
+            try
+            {
+                WriteLine($"Adding translations to language {name}");
+
+                if (TryGetLanguage(name, out var customLanguage))
+                {
+                    TextTranslationPatches.AddNewTranslation(customLanguage, regularEntries, shipLogEntries, uiEntries);
+                    return;
+                }
+                WriteError($"The custom language {name} isn't registered yet");
+
+            }
+            catch (Exception ex)
+            {
+                WriteError($"Failed to add translation to language {name}. {ex}");
+            }
+        }
+        public void AddRegularTranslation(string name, string commonKeyPrefix, params string[] entries)
```

Comment:
Why do we need two different versions of each method here?

Reply:
The idea is that you should use `public void AddRegularTranslation(string name, string commonKeyPrefix, params string[] entries)` only if they have the same prefix for the key of all the entries, like the translation entries from sign texts. And `public void AddRegularTranslation(string languageName, params KeyValuePair<string, string>[] entries)` if you are getting the translations from somewhere where the key prefix can be different for each entry, so loading from files is easier. It is supposed to work as a version of `public void AddTranslation(string languageName, KeyValuePair<string, string>[] regularEntries, KeyValuePair<string, string>[] shipLogEntries, KeyValuePair<int, string>[] uiEntries)` but where you only care about the regular translation table.